src/main/java/strategy/OperationStrategyFactory.java

```diff
@@ -0,0 +1,21 @@
+package strategy;
+
+import java.util.HashMap;
+import java.util.Map;
+import model.FruitTransaction;
+
+public class OperationStrategyFactory {
```

Comment:
Strategy itself is pattern to provide some implementation, no need to add Factory to naming
```suggestion
public class OperationStrategy {
```